js/modules/k6/crypto/x509/x509.go

```diff
@@ -0,0 +1,274 @@
+	"crypto/dsa"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/sha1" // #nosec G505
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"github.com/loadimpact/k6/js/common"
+	"github.com/pkg/errors"
+)
+
+// X509 certificate functionality
+type X509 struct{}
+
+// Certificate is an X.509 certificate
+type Certificate struct {
+	Subject            Subject
+	Issuer             Issuer
+	NotBefore          string    `js:"notBefore"`
+	NotAfter           string    `js:"notAfter"`
+	AltNames           []string  `js:"altNames"`
+	SignatureAlgorithm string    `js:"signatureAlgorithm"`
+	FingerPrint        []byte    `js:"fingerPrint"`
+	PublicKey          PublicKey `js:"publicKey"`
+}
+
+// RDN is a component of an X.509 distinguished name
+type RDN struct {
+	Type  string
+	Value string
+}
+
+// Subject is a certificate subject
+type Subject struct {
+	CommonName             string `js:"commonName"`
+	Country                string
+	PostalCode             string   `js:"postalCode"`
+	StateOrProvinceName    string   `js:"stateOrProvinceName"`
+	LocalityName           string   `js:"localityName"`
+	StreetAddress          string   `js:"streetAddress"`
+	OrganizationName       string   `js:"organizationName"`
+	OrganizationalUnitName []string `js:"organizationalUnitName"`
+	Names                  []RDN
+}
+
+// Issuer is a certificate issuer
+type Issuer struct {
+	CommonName          string `js:"commonName"`
+	Country             string
+	StateOrProvinceName string `js:"stateOrProvinceName"`
+	LocalityName        string `js:"localityName"`
+	OrganizationName    string `js:"organizationName"`
+	Names               []RDN
+}
+
+// PublicKey is used for decryption and signature verification
+type PublicKey struct {
+	Algorithm string
+	Key       interface{}
+}
+
+// New constructs the X509 interface
+func New() *X509 {
+	return &X509{}
+}
+
+// Parse produces an entire X.509 certificate
+func (X509) Parse(ctx context.Context, encoded string) Certificate {
+	parsed, err := parseCertificate(encoded)
+	if err != nil {
+		throw(ctx, err)
+	}
+	certificate, err := makeCertificate(parsed)
+	if err != nil {
+		throw(ctx, err)
+	}
+	return certificate
+}
+
+// GetAltNames extracts alt names
+func (X509) GetAltNames(ctx context.Context, encoded string) []string {
+	parsed, err := parseCertificate(encoded)
+	if err != nil {
+		throw(ctx, err)
+	}
+	return altNames(parsed)
+}
+
+// GetIssuer extracts certificate issuer
+func (X509) GetIssuer(ctx context.Context, encoded string) Issuer {
+	parsed, err := parseCertificate(encoded)
+	if err != nil {
+		throw(ctx, err)
+	}
+	return makeIssuer(parsed.Issuer)
+}
+
+// GetSubject extracts certificate subject
+func (X509) GetSubject(ctx context.Context, encoded string) Subject {
+	parsed, err := parseCertificate(encoded)
+	if err != nil {
+		throw(ctx, err)
+	}
+	return makeSubject(parsed.Subject)
+}
+
+func parseCertificate(encoded string) (*x509.Certificate, error) {
+	decoded, _ := pem.Decode([]byte(encoded))
```

Comment:
Maybe having `encoded` be `[]byte` from the start will be better ?

Reply:
Hmm I'm not 100% sure, but I think the JS code will work absolutely the same way if we had `[]byte` parameters instead of `string`.

Reply:
Guess it doesn't hurt anything. Made the change.